### PR TITLE
[Feature] [SR SQL Planner Summer Camp] add showRestore analyzer

### PR DESCRIPTION
**Pull request by melt-code**
## What type of PR is this：
- [ ] bug
- [x] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #7951

## Problem Summary(Required) ：

add showRestore analyzer

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 30 / 38 (78.95%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|1|00.00%|[418]|
|:large_blue_circle:|com/starrocks/analysis/ShowRestoreStmt.java|1|4|25.00%|[100, 101, 123]|
|:large_blue_circle:|com/starrocks/sql/analyzer/ShowRestoreAnalyzer.java|21|25|84.00%|[20, 40, 41, 53]|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|6|6|100.00%|[]|

**Comment by melt-code** [comment]
run starrocks_fe_unittest

**Comment by melt-code** [comment]
run starrocks_fe_unittest

**Comment by melt-code** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 32 / 38 (84.21%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|1|00.00%|[418]|
|:large_blue_circle:|com/starrocks/analysis/ShowRestoreStmt.java|2|4|50.00%|[101, 123]|
|:large_blue_circle:|com/starrocks/sql/analyzer/ShowRestoreAnalyzer.java|22|25|88.00%|[20, 41, 53]|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|6|6|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 36 / 42 (85.71%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|1|00.00%|[443]|
|:large_blue_circle:|com/starrocks/analysis/ShowRestoreStmt.java|4|6|66.67%|[105, 127]|
|:large_blue_circle:|com/starrocks/sql/analyzer/ShowRestoreAnalyzer.java|19|22|86.36%|[18, 39, 52]|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|6|6|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|5|5|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4 [hunk 1]
```diff
@@ -136,6 +136,9 @@ statement
 
     // proc
     | showProcStatement                                                                      #showProc
+
+    // Backup Store Satement
```
Comment (Youngwb) [inline fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4:140]: wrong comment

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java [hunk 2]
```diff
@@ -2380,6 +2381,16 @@ public ParseNode visitShowProcesslistStatement(StarRocksParser.ShowProcesslistSt
         return new ShowProcesslistStmt(isShowFull);
     }
 
+    // ------------------------------------------- Backup Store Statement ----------------------------------------------
```
Comment (Youngwb) [inline fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java:2384]:
```suggestion
    // ------------------------------------------- Backup Restore Statement ----------------------------------------------
```

File: fe/fe-core/src/main/java/com/starrocks/sql/ast/AstVisitor.java [hunk 3]
```diff
@@ -438,6 +439,10 @@ public R visitShowHistogramStatsMetaStatement(ShowHistogramStatsMetaStmt stateme
         return visitShowStatement(statement, context);
     }
 
+    public R visitShowRestoreStmt(ShowRestoreStmt statement, C context) {
+        return visitStatement(statement, context);
```
Comment (Youngwb) [inline fe/fe-core/src/main/java/com/starrocks/sql/ast/AstVisitor.java:443]:
```suggestion
        return visitShowStatement(statement, context);
```

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4 [hunk 4]
```diff
@@ -836,6 +839,11 @@ showProcStatement
     : SHOW PROC path=string
     ;
 
+// ---------------------------------------- Backup Store Statement -----------------------------------------------------
+showRestoreStatement
+    : SHOW RESTORE (FROM identifier)?
```
Comment (Youngwb) [inline fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4:844]: miss where clause?

Comment (melt-code) [reply fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4:844]: I see there is no where clause in the official document.https://docs.starrocks.com/zh-cn/main/sql-reference/sql-statements/data-manipulation/SHOW%20RESTORE

Comment (Youngwb) [reply fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4:844]: The documentation is not comprehensive and you should follow the source code， you can test the where clause using old parser and analyzer

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 36 / 44 (81.82%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|1|00.00%|[456]|
|:large_blue_circle:|com/starrocks/analysis/ShowRestoreStmt.java|4|6|66.67%|[103, 125]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|6|8|75.00%|[2430, 2431]|
|:large_blue_circle:|com/starrocks/sql/analyzer/ShowRestoreAnalyzer.java|19|22|86.36%|[18, 39, 52]|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|5|5|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 36 / 44 (81.82%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|1|00.00%|[456]|
|:large_blue_circle:|com/starrocks/analysis/ShowRestoreStmt.java|4|6|66.67%|[103, 125]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|6|8|75.00%|[2430, 2431]|
|:large_blue_circle:|com/starrocks/sql/analyzer/ShowRestoreAnalyzer.java|19|22|86.36%|[18, 39, 52]|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|5|5|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4 (hunk 1 above)
Comment (melt-code) [reply fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4]: It has been modified

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java (hunk 2 above)
Comment (melt-code) [reply fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java:2422]: It has been modified

File: fe/fe-core/src/main/java/com/starrocks/sql/ast/AstVisitor.java (hunk 3 above)
Comment (melt-code) [reply fe/fe-core/src/main/java/com/starrocks/sql/ast/AstVisitor.java]: It has been modified

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4 (hunk 4 above)
Comment (melt-code) [reply fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4]: It has been modified

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 36 / 44 (81.82%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|1|00.00%|[422]|
|:large_blue_circle:|com/starrocks/analysis/ShowRestoreStmt.java|4|6|66.67%|[103, 125]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|6|8|75.00%|[2430, 2431]|
|:large_blue_circle:|com/starrocks/sql/analyzer/ShowRestoreAnalyzer.java|19|22|86.36%|[18, 38, 51]|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|5|5|100.00%|[]|

**Review by Youngwb — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 36 / 44 (81.82%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|1|00.00%|[422]|
|:large_blue_circle:|com/starrocks/analysis/ShowRestoreStmt.java|4|6|66.67%|[103, 125]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|6|8|75.00%|[2430, 2431]|
|:large_blue_circle:|com/starrocks/sql/analyzer/ShowRestoreAnalyzer.java|19|22|86.36%|[18, 38, 51]|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|5|5|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 34 / 42 (80.95%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|1|00.00%|[468]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|4|6|66.67%|[2835, 2836]|
|:large_blue_circle:|com/starrocks/analysis/ShowRestoreStmt.java|4|6|66.67%|[103, 125]|
|:large_blue_circle:|com/starrocks/sql/analyzer/ShowRestoreAnalyzer.java|19|22|86.36%|[18, 38, 51]|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|5|5|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 85 / 105 (80.95%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|1|00.00%|[480]|
|:large_blue_circle:|com/starrocks/analysis/ShowRestoreStmt.java|3|6|50.00%|[102, 103, 125]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|4|6|66.67%|[2941, 2942]|
|:large_blue_circle:|com/starrocks/sql/analyzer/AnalyzerUtils.java|50|63|79.37%|[327, 346, 351, 357, 373, 374, 382, 389, 398, 409, 410, 419, 425]|
|:large_blue_circle:|com/starrocks/sql/analyzer/ShowRestoreAnalyzer.java|9|10|90.00%|[9]|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/BackupStmtAnalyzer.java|12|12|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 23 / 25 (92.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|1|00.00%|[445]|
|:large_blue_circle:|com/starrocks/analysis/ShowRestoreStmt.java|5|6|83.33%|[124]|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|6|6|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/BackupRestoreAnalyzer.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|6|6|100.00%|[]|

**Comment by melt-code** [comment]
run starrocks_admit_test

**Review by HangyuanLiu — APPROVED** [review]

**Review by imay — APPROVED** [review]

**Pull request merged**